python/tskit/trees.py

```diff
@@ -1267,12 +1267,36 @@ def num_roots(self):
         The number of roots in this tree, as defined in the :attr:`~Tree.roots`
         attribute.
 
-        Requires O(number of roots) time.
+        Only requires O(number of roots) time.
 
         :rtype: int
         """
         return self._ll_tree.get_num_roots()
 
+    @property
+    def has_single_root(self):
+        """
+        Only ``True`` if this tree has a single root.
```

Comment:
I would say something like "True if this tree has a single root and equivalent to ``tree.num_roots == 1``. This is a O(1) operation."

Good to be precise and also reassure people that it's an efficient operation.

Reply:
I guess the "Only" makes it clear here that e.g. we don't return False for num_roots == 0. It's an "if and only if". Probably overly pedantic.

Reply:
Maybe "False otherwise" reads better. I'll change it.